Documentation/building/windows-instructions.md

```diff
@@ -8,44 +8,81 @@ MSBuild version 15 or later is required.
 
 # Building Xamarin.Android
 
- 1. Install the [build dependencies](dependencies.md).
+ 1. Run the [Visual Studio Installer](https://visualstudio.microsoft.com/vs/).
 
- 2. Clone the xamarin-android repo:
+ 2. Under the **Workloads** tab, ensure that the **Mobile development with
+    .NET** workload is installed.  Under the **Optional** items for the
+    workload, ensure **Android SDK setup** is selected.
+
+ 3. Also ensure the **.NET desktop development** workload is installed.  Under
+    the **Optional** items for the workload, ensure the following items are
+    installed:
+
+      * **.NET Framework 4 &ndash; 4.6 development tools**
+      * **.NET Framework 4.6.2 development tools**
+      * **.NET Framework 4.7 development tools**
+      * **.NET Core 2.0 development tools**
+
+    The following items are also recommended:
+
+      * **.NET Framework 4.7.1 development tools**
+      * **.NET Framework 4.7.2 development tools**
+
+ 4. Under the **Individual components** tab, ensure that **Code tools > Git for
+    Windows** is installed.
+
+ 5. Ensure the .NET Framework 3.5 SP1 Runtime is installed by downloading and
+    running the installer from
+    <https://www.microsoft.com/net/download/visual-studio-sdks>.
+
+ 6. Download and install the Java SE 8 JDK from the [Oracle
+    website][oracle-jdk].  You can use either the Windows x64 or Windows x86
+    version.  Make sure **Development Tools** is selected for installation when
+    running the installer.
+
+ 7. Clone the xamarin-android repo:
 
         git clone https://github.com/xamarin/xamarin-android.git
 
- 3. Navigate to the `xamarin-android` directory
+ 8. Navigate to the `xamarin-android` directory
 
- 4. (Optional) [Configure the build](configuration.md).
+ 9. (Optional) [Configure the build](configuration.md).
 
- 5. Prepare the project:
+10. In a [Developer Command Prompt][developer-prompt], prepare the project:
```

Comment:
So another option is to just put `MSBuild.exe` in `%PATH%`, which is what I do. I use powershell and only resort to the developer command prompt for something really specific.

Maybe I could add a quick explanation on how to do this in a future PR.

Reply:
Sounds good. The related trick I've been using is to start a Developer Command Prompt and then run `powershell` as the first command in it so that I can use PowerShell commands and ReadLine handling but inherit the environment from the Developer Command Prompt. I've occasionally taken advantage of that approach to switch between the Release and Preview MSBuild versions, for example. I also sometimes use `devenv /edit` to open files for editing, so it's handy to have that in the path too.